src/VisualStudio/Core/Def/SampleToolWindow/DocumentOutlineHelper.cs

```diff
@@ -0,0 +1,129 @@
+                    var documentSymbol = body[i];
+                    var ds = new DocSymbol(
+                        documentSymbol.Name,
+                        documentSymbol.Kind,
+                        documentSymbol.Range.Start.Line,
+                        documentSymbol.Range.Start.Character,
+                        documentSymbol.Range.End.Line,
+                        documentSymbol.Range.End.Character);
+                    var children = documentSymbol.Children;
+                    if (children is not null)
+                    {
+                        ds = AddNodes(ds, children);
+                    }
+
+                    docSymbols.Add(ds);
+                }
+            }
+
+            docSymbols = docSymbols.OrderBy(x => x.StartLine).ThenBy(x => x.StartChar).ToList();
+            for (var i = 0; i < docSymbols.Count; i++)
+            {
+                docSymbols[i].Children = Sort(docSymbols[i].Children, SortOption.Order);
+            }
+
+            return docSymbols;
+        }
+
+        internal static DocSymbol AddNodes(DocSymbol newNode, DocumentSymbol[] children)
+        {
+            var newChildren = new ObservableCollection<DocSymbol>();
+
+            if (children is null || children.Length == 0)
+            {
+                return newNode;
+            }
+            else
+            {
+                for (var i = 0; i < children.Length; i++)
+                {
+                    var child = children[i];
+                    var newChild = new DocSymbol(
+                        child.Name,
+                        child.Kind,
+                        child.Range.Start.Line,
+                        child.Range.Start.Character,
+                        child.Range.End.Line,
+                        child.Range.End.Character);
+                    if (child.Children is not null)
+                    {
+                        newChild = AddNodes(newChild, child.Children);
+                    }
+
+                    newChildren.Add(newChild);
+                }
+
+                newNode.Children = newChildren;
+                return newNode;
+            }
+        }
+
+        internal static ObservableCollection<DocSymbol> Sort(ObservableCollection<DocSymbol> docSymbols, SortOption sortOption)
+        {
+            if (docSymbols.Count == 0)
+            {
+                return docSymbols;
+            }
+
+            var result = new List<DocSymbol>();
+            switch (sortOption)
+            {
+                case SortOption.Name:
+                    result = docSymbols.OrderBy(x => x.Name).ToList();
+                    break;
+                case SortOption.Order:
+                    result = docSymbols.OrderBy(x => x.StartLine).ThenBy(x => x.StartChar).ToList();
+                    break;
+                case SortOption.Type:
+                    result = docSymbols.OrderBy(x => x.SymbolKind).ThenBy(x => x.Name).ToList();
+                    break;
+            }
```

Comment:
```suggestion
            var result = switch sortOption
            {
	            SortOption.Name => docSymbols.OrderBy(x => x.Name).ToList(),
	            SortOption.Order => docSymbols.OrderBy(x => x.StartLine).ThenBy(x => x.StartChar).ToList(),
	            SortOption.Type => docSymbols.OrderBy(x => x.SymbolKind).ThenBy(x => x.Name).ToList()
            };
```

Reply:
ToList could possibly be deduplicated outside the switch expression as well.

Reply:
Done, and I removed ToList() altogether - we can just keep it as an `ObservableCollection<DocumentSymbolViewModel>`